src/compute-client/src/plan/join/linear_join.rs

```diff
@@ -309,8 +313,8 @@ impl LinearJoinPlan {
         // Form and return the complete join plan.
         let plan = LinearJoinPlan {
             source_relation,
-            source_key,
-            initial_closure,
+            source_key: source_arrangement.map(|(key, _, _)| key.clone()),
+            initial_closure: None,
```

Comment:
So `initial_closure` will always be `None`? In that case, can we also delete [a bunch of code from the rendering](https://github.com/MaterializeInc/materialize/blob/50bdf26fac2bb7cf3dacdcc6e85624b53fb6c60c/src/compute/src/render/join/linear_join.rs#L91)?

Reply:
I vote against removing it. There isn't a contract that it will always be `None`, it just happens that at the moment there is that behavior. If we change this code back for some reason, and end up with a quietly incorrect renderer, that seems  worse.